janus_core/calculations/phonons.py

```diff
@@ -191,50 +208,90 @@ def calc_phonons(self, write_results: bool = True) -> None:
         ]
 
         phonon.produce_force_constants()
-        phonon.run_mesh()
         self.results["phonon"] = phonon
 
         if self.logger:
             self.logger.info("Phonons calculation complete")
 
         if write_results:
-            self.write_phonons()
+            self.write_force_constants(force_consts_to_hdf5=self.hdf5)
+
+    def calc_bands(self, write_results: bool = True) -> None:
+        """
+        Calculate band structure and optionally write and plot results.
 
-    def write_phonons(
+        Parameters
+        ----------
+        write_results : bool
+            Whether to write out results to file. Default is True.
+        """
+        # Calculate phonons is not already run
+        if "phonon" not in self.results:
+            self.calc_force_constants(write_results=False)
+        self.write_band_structure(write_bands=write_results)
+
+    def write_band_structure(
         self,
         *,
-        params_file: Optional[PathLike] = None,
+        write_bands: bool = None,
         bands_file: Optional[PathLike] = None,
-        force_consts_to_hdf5: bool = False,
-        force_consts_file: Optional[PathLike] = None,
+        plot_file: Optional[PathLike] = None,
     ) -> None:
         """
-        Write results of phonon calculations.
+        Write results of band structure calculations.
 
         Parameters
         ----------
-        params_file : Optional[PathLike]
-            Name of yaml file to save phonon parameters. Default is inferred from
-            `file_prefix`.
+        write_bands : bool
+            Whether to write out results to file. Default is True.
         bands_file : Optional[PathLike]
             Name of yaml file to save band structure. Default is inferred from
             `file_prefix`.
+        plot_file : Optional[PathLike]
+            Name of svg file to save band structure. Default is inferred from
+            `file_prefix`.
+        """
+
+        bands_file = self._set_filename("auto_bands.yml", bands_file)
+        self.results["phonon"].auto_band_structure(
+            write_yaml=write_bands, filename=bands_file
+        )
+        if self.plot:
```

Comment:
What about plot to screen? If you say "plot", that implies to screen to me.

Reply:
will not plot to screen, the ascii plot is cool but... but to file is nicer I will rename to plot_to_file